File: HeartForCharity/HeartForCharity.Services/Database/UserProfile.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Net;

namespace HeartForCharity.Services.Database
{
    public class UserProfile
    {
        [Key]
        public int UserProfileId { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        public int UserId { get; set; }

        [Required]
        [MaxLength(100)]
        public string FirstName { get; set; } = null!;

        [Required]
        [MaxLength(100)]
        public string LastName { get; set; } = null!;

        [MaxLength(20)]
        public string? PhoneNumber { get; set; }

        public DateTime? DateOfBirth { get; set; }

        [MaxLength(500)]
        public string? ProfilePictureUrl { get; set; }

        [ForeignKey(nameof(Address))]
        public int? AddressId { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? DeletedAt { get; set; }

        // Navigacijska svojstva
        public virtual User User { get; set; } = null!;
        public virtual Address? Address { get; set; }
        public virtual ICollection<VolunteerSkill> VolunteerSkills { get; set; } = new List<VolunteerSkill>();
        public virtual ICollection<Donation> Donations { get; set; } = new List<Donation>();
        public virtual ICollection<VolunteerApplication> VolunteerApplications { get; set; } = new List<VolunteerApplication>();
        public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();
        public virtual ICollection<Recommendation> Recommendations { get; set; } = new List<Recommendation>();
        public virtual ICollection<Notification> Notifications { get; set; } = new List<Notification>(); // Dodato
    }
}

File: HeartForCharity/UI/heartforcharity_mobile/lib/model/responses/user_profile.dart
class UserProfile {
  final int userProfileId;
  final int userId;
  final String firstName;
  final String lastName;
  final String? phoneNumber;
  final DateTime? dateOfBirth;
  final String? profilePictureUrl;
  final int? addressId;
  final String? cityName;
  final String? countryName;
  final DateTime? createdAt;

  UserProfile({
    this.userProfileId = 0,
    this.userId = 0,
    this.firstName = '',
    this.lastName = '',
    this.phoneNumber,
    this.dateOfBirth,
    this.profilePictureUrl,
    this.addressId,
    this.cityName,
    this.countryName,
    this.createdAt,
  });

  factory UserProfile.fromJson(Map<String, dynamic> json) => UserProfile(
        userProfileId: json['userProfileId'] ?? 0,
        userId: json['userId'] ?? 0,
        firstName: json['firstName'] ?? '',
        lastName: json['lastName'] ?? '',
        phoneNumber: json['phoneNumber'],
        dateOfBirth: json['dateOfBirth'] != null ? DateTime.parse(json['dateOfBirth']) : null,
        profilePictureUrl: json['profilePictureUrl'],
        addressId: json['addressId'],
        cityName: json['cityName'],
        countryName: json['countryName'],
        createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      );
}
